lib/elixir/lib/calendar/iso.ex

```diff
@@ -421,6 +436,21 @@ defmodule Calendar.ISO do
     parse_formatted_date(year, month, day, multiplier)
   end
 
+  defp do_parse_date(<<y, _rest::binary>> = bin, multiplier, :extended) when is_digit_char(y) do
+    case Integer.parse(bin) do
+      {year, <<@ext_date_sep, unquote(match_month_day)>>}
+      when is_big_year(year) and unquote(guard_month_day) ->
+        {month, day} = unquote(read_month_day)
+        parse_formatted_date(year, month, day, multiplier)
+
+      :error ->
```

Comment:
Wasn't sure about making the last two clauses into a simple `_ -> {:error, :invalid_format}`.